### PR TITLE
[SINGA-268] Add IPython notebooks to the documentation

### PR DESCRIPTION
**Pull request by moazreyad**

### CODE REVIEW

File: doc/notebook/rnn.ipynb [hunk 1]
```diff
@@ -0,0 +1,151 @@
+{
```
Comment (nudles) [inline doc/notebook/rnn.ipynb]: RNN training requires cudnn5 and the prediction requires the pre-trained model parameters, which is not a small file and may not be suitable to put it into github.
I suggest we remove rnn and vgg. There are examples about rnn and vgg in examples/ folder (no notebooks).
I think, we can add one notebook to show the usage of tensor, initializer, optimizer, devices, and other components before regression/rbm/mlp.

Comment (moazreyad) [inline doc/notebook/rnn.ipynb]: RNN and VGG are removed.

**Comment by moazreyad** [comment]
I checked the notebook and they work fine. I made small fixes and merged all changes in one commit.

**Comment by nudles** [comment]
I moaz, please check again.
You can add new commits (no need to overwrite the existing commits).

**Comment by moazreyad** [comment]
I got an error:

 43     print 'Epoch %d, Reconstruction error per image = %f' % (epoch, err / num_train_batch / batch_size)

NameError: name 'num_train_batch' is not defined

in rbm.ipynb

**Comment by moazreyad** [comment]
Another problem: regression.ipynb points to the Tensor example (http://singa.incubator.apache.org/en/docs/tensor.html) which is not working. It gives an error: 

NameError: name 'np' is not defined

I think the tensor example in the documentation needs to be updated.

**Pull request merged**